articles/115_topic_and_service_name_mapping.md

```diff
@@ -0,0 +1,317 @@
+
+Additionally, DDS has a hard limit on topic names of 256 characters, so an additional goal is to minimize the number of extra characters used when mapping from ROS to DDS names.
+
+## ROS 2 Topic and Service Name Constraints
+
+In this section an outline of the proposed constrains for ROS 2 topic and service names will be enumerated along with rationales where appropriate.
+
+For convenience here is a summary of all rules for topic and service names in ROS 2:
+
+- may contain alphanumeric characters (`[0-9|a-z|A-Z]`), underscores (`_`), tildes (`~`), or forward slashes (`/`)
+- must not start with a numeric character (`[0-9]`)
+- must not contain any number of repeated underscores (`_`)
+- must not end with an underscore (`_`)
+- must not have an underscore (`_`) proceeded by a forward slash (`/`), i.e. `_/`
+- must not end with a forward slash (`/`)
+- must not contain any number of repeated forward slashes (`/`)
+- must not have tilde (`~`) adjacent to anything other than forward slashes (`/`)
+- may start with or end with a tilde (`~`)
+
+Additionally, topic and service names can be represented in the [Uniform Resource Locator (URL)](https://en.wikipedia.org/wiki/Uniform_Resource_Locator) format to further disambiguate the resource name.
+A topic name may be preceded by a `rostopic://` scheme prefix, and a service name may be preceded by a `rosservice://` scheme prefix.
+For example, the absolute name `/foo` could also be represented as a topic with `rostopic:///foo` or as a service with `rosservice:///foo`.
+Note the triple forward slash (`/`), which is a similar style as the `file://` scheme.
+A relative name `foo/bar` could would be represented (as a topic) with `rostopic://foo/bar`.
+
+### ROS 2 Name Examples
+
+For example, these are valid names:
+
+| `foo`       | `abc123`            | `_foo`               | `Foo` | `BAR`       |
+| `~`         | `foo/bar`           | `~/foo`              | `/~`  | `foo/~/bar` |
+| `foo/_/bar` | `rosservice:///foo` | `rostopic://foo/bar` |       |             |
+
+But these are not valid names:
+
+| `123abc`   | `123`  | `__foo`  | `foo__bar` | `foo bar`  |
+| `foo__`    | ` `    | `foo_`   | `foo//bar` | `foo/`     |
+| `foo_/bar` | `~foo` | `foo~`   | `foo~/bar` | `foo/~bar` |
+| `/_/bar`   | `_`    | `_/_bar` |            |            |
+
+### Namespaces
+
+Topic and service names:
+
+- may be split into tokens using a forward slash (`/`) as a delimiter
+- must not end with a forward slash (`/`)
+
+The last token is the topic or service name, and any preceding tokens comprise the namespace of the topic or service.
+
+For example, the topic name `/foo/bar/baz` comprises of a topic or service named `baz` in the `/foo/bar` namespace.
+In another example, the name `/foo` does not use splitting into tokens, so it comprises of a topic or service named `foo` in the `/` namespace (the root namespace).
```

Comment:
The wording here could be confusing. '/foo' can be split into tokens, we just find that there are no additional tokens (parent names) between the root and the topic name, so it is in the root namespace.